src/utils/transaction.ts

```diff
@@ -314,6 +586,203 @@ export const buildCrossChainAction = async (
     }
   }
 
+  if (
+    transactionBlock.type === TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE &&
+    !!transactionBlock?.values?.fromChainId &&
+    !!transactionBlock?.values?.fromAsset &&
+    !!transactionBlock?.values?.amount &&
+    !!transactionBlock?.values?.offer &&
+    !!transactionBlock?.values?.receiverAddress &&
+    !!transactionBlock?.values?.toAsset
+  ) {
+    if (!transactionBlock?.values?.hasEnoughPLR) {
```

Comment:
I think I left comment earlier that it does not read well as you're checking negative value first against positive expectation

you can also avoid `else` by just returning from previous `if` and code won't proceed – saves spaces and reads easier